python/pyspark/sql/session.py

```diff
@@ -611,8 +611,8 @@ def _inferSchema(
         :class:`pyspark.sql.types.StructType`
         """
         first = rdd.first()
-        if not first:
-            raise ValueError("The first row in RDD is empty, " "can not infer schema")
+        if first is None:
```

Comment:
Hm, can this be `None`? Seems like `if rs:` check `RDD.first`  will always throw an exception in case this is `None`. Can we just check if this has a length and larger than `0`?

Reply:
Sounds good! Updated.

Just to confirm, we may not want to support ` spark.createDataFrame(spark._sc.parallelize([[], []]))` for now, is that right?

Reply:
If `rdd.first()` uses `if rs:` check, `first` will never be `[]`, `{}`, `0` or `""`, either?
I guess we should fix `rdd.first()` to properly return the first row, and `if first is None:` check here is vaild according to the error message `"The first row in RDD is empty, can not infer schema"`.

Reply:
The `first` can be `None`.

```py
>>> print(sc.parallelize([None, 2, 3, 4]).first())
None
```

We should also check if `first is None` because `None` is not `Sized`.

```py
>>> isinstance(None, Sized)
False
```

Also we need a test to check the first row is `None` and it fails.

Reply:
It cannot infer the type from `None` or `""`. It has to be either a list, dictionary or `Row` instance.